test/token-transfer-binary-regulator.js

```diff
@@ -0,0 +1,128 @@
+      );
+    });
+
+    it("Shouldn't allow me to cancel someone else's transfer", async () => {
+      await regulator.requestTransfer(ADDRESS_1, ADDRESS_2, token.address, 500, { from: ADDRESS_1 });
+      const transferCount = await regulator.transferCount();
+      await checkErrorRevert(regulator.invalidateRequest(transferCount - 1), "colony-token-regulator-not-from-address");
+    });
+
+    it("Should allow me to cancel a transfer I proposed", async () => {
+      await regulator.requestTransfer(ADDRESS_1, ADDRESS_2, token.address, 500, { from: ADDRESS_1 });
+      const transferCount = await regulator.transferCount();
+      await regulator.invalidateRequest(transferCount - 1, { from: ADDRESS_1 });
+      const transfer = await regulator.transfers(transferCount - 1);
+      assert.equal(transfer.valid, false);
+    });
+
+    it("Should allow an admin to decline a transfer", async () => {
+      await regulator.requestTransfer(ADDRESS_1, ADDRESS_2, token.address, 500, { from: ADDRESS_1 });
+      const transferCount = await regulator.transferCount();
+
+      let transfer = await regulator.transfers(transferCount - 1);
+      assert.equal(transfer.valid, true);
+
+      const txData = await regulator.contract.methods.invalidateRequest(transferCount - 1).encodeABI();
+      await colonyMultiSig.submitTransaction(regulator.address, 0, txData);
+
+      transfer = await regulator.transfers(transferCount - 1);
+      assert.equal(transfer.valid, false);
+    });
+
+    it("should not allow a transfer to be approved more than once", async () => {
+      let txData = await token.contract.methods.mint(1000).encodeABI();
+      await colonyMultiSig.submitTransaction(token.address, 0, txData);
+      txData = await token.contract.methods.transfer(ADDRESS_1, 1000).encodeABI();
+      await colonyMultiSig.submitTransaction(token.address, 0, txData);
+
+      // Make request to regulator
+      await token.approve(regulator.address, 1000, { from: ADDRESS_1 });
+      await regulator.requestTransfer(ADDRESS_1, ADDRESS_2, token.address, 500, { from: ADDRESS_1 });
+
+      const transferCount = await regulator.transferCount();
+
+      // Approve request through multisig
+      txData = await regulator.contract.methods.executeTransfer(transferCount - 1).encodeABI();
+      await colonyMultiSig.submitTransaction(regulator.address, 0, txData);
+
+      const a1BalanceBefore = await token.balanceOf(ADDRESS_1);
+      const a2BalanceBefore = await token.balanceOf(ADDRESS_2);
+
+      await expectEvent(colonyMultiSig.submitTransaction(regulator.address, 0, txData), "ExecutionFailure");
+
+      // Double check balances didn't change
+      const a1BalanceAfter = await token.balanceOf(ADDRESS_1);
+      const a2BalanceAfter = await token.balanceOf(ADDRESS_2);
+
+      assert.equal(a1BalanceAfter.toString(), a1BalanceBefore.toString());
+      assert.equal(a2BalanceAfter.toString(), a2BalanceBefore.toString());
+    });
+  });
```

Comment:
Since transfers aren't cumulative, can we add a test of how this behaves with multiple transfers of the same account but a single approved budget in the ERC20.